pkg/wekafs/mounter.go

```diff
@@ -79,14 +82,55 @@ func (m *wekaMount) doUnmount() error {
 	return err
 }
 
-func (m *wekaMount) doMount() error {
+func (m *wekaMount) doMount(apiClient *apiclient.ApiClient) error {
 	glog.Infof("Creating mount for filesystem %s on mount point %s", m.fsRequest.fs, m.mountPoint)
+	mountToken := ""
 	if err := os.MkdirAll(m.mountPoint, DefaultVolumePermissions); err != nil {
 		return err
 	}
 	if m.debugPath == "" {
-		glog.V(3).Infof("Calling k8s mounter for fs: %s (xattr %t) @ %s", m.fsRequest.fs, m.fsRequest.xattr, m.mountPoint)
-		return m.kMounter.Mount(m.fsRequest.fs, m.mountPoint, "wekafs", getMountOptions(m.fsRequest))
+		if apiClient == nil {
+			glog.V(3).Infof("No API client for mount, not requesting mount token")
+		} else {
+			if !apiClient.SupportsAuthenticatedMounts() {
+				glog.V(3).Infof("API client not supports authenticated mounts, skipping")
+			} else {
+				glog.V(3).Infof("Requesting mount token for filesystem %s via API", m.fsRequest.fs)
+				filesystem, err := apiClient.GetFileSystemByName(m.fsRequest.fs)
+				if err != nil {
+					return err
+				}
+				if filesystem != nil {
+					req := &apiclient.FileSystemMountTokenRequest{Uid: filesystem.Uid}
+					token := &apiclient.FileSystemMountToken{}
```

Comment:
why not just create and return in function with cleaner signature, instead of passing references (and better to avoid use of pointers, when not really required. Does not seem really required to me)

Reply:
Made for retain signature similarity with other ApiClient functions. Unless I don't get you right.

Pointer required for cases where this is not supported, the generic mount function will either get a valid token or nil

Reply:
but this is not generic function
it's `apiClient` function that returns struct that also part of apiclient
it would make more sense just to create and return token
passing pointer makes sense for generalistion when it's not specific struct, but interfaces. not a case here